# Google Trends Data Analysis

Datasets used and sources:

- Bitcoin Search Trend CSV (Google Trends)
- Daily Bitcoin Price CSV (Yahoo Finance)

- TESLA Search Trend vs Price (Google Trends, Yahoo Finance)
  1st June 2010 - 1st June 2020  
- UE Benefits Search vs UE Rate 2004-19 (FRED)
- UE Benefits Search vs UE Rate 2004-20 (FRED) 

In [11]:
import pandas as pd
import matplotlib.pyplot as plt

## TESLA Search Trend vs Price

### Data Exploration

In [13]:
tesla = pd.read_csv("data/TESLA Search Trend vs Price.csv")

In [39]:
tesla

MONTH  TSLA_WEB_SEARCH  TSLA_USD_CLOSE
0    2010-06-01                3        4.766000
1    2010-07-01                3        3.988000
2    2010-08-01                2        3.896000
3    2010-09-01                2        4.082000
4    2010-10-01                2        4.368000
..          ...              ...             ...
119  2020-05-01               16      167.000000
120  2020-06-01               17      215.962006
121  2020-07-01               24      286.152008
122  2020-08-01               23      498.320007
123  2020-09-01               31      407.339996

[124 rows x 3 columns]

In [41]:
tesla.dtypes

MONTH               object
TSLA_WEB_SEARCH      int64
TSLA_USD_CLOSE     float64
dtype: object

In [43]:
tesla.describe()

TSLA_WEB_SEARCH  TSLA_USD_CLOSE
count       124.000000      124.000000
mean          8.725806       50.962145
std           5.870332       65.908389
min           2.000000        3.896000
25%           3.750000        7.352500
50%           8.000000       44.653000
75%          12.000000       58.991999
max          31.000000      498.320007

### Data Exploration Summary

- There are 3 columns with 124 rows of data
- The column data types are: 'MONTH' = Object, 'TSLA_WEB_SEARCH' = int64, 'TSLA_USD_CLOSE' = float64
- The data in the columns represent:
  'MONTH' = The date, 'TSLA_WEB_SEARCH' = Monthly search interest on Google, 'TSLA_USD_CLOSE' = Closing Price of TSLA stock in USD
- The highest number of searches during the period was 31
- The lowest number of searches during the period was 2
- The highest Closing Price of TSLA stock during the period was \\$498.320007
- The lowest Closing Price of TSLA stock during the period was \\$3.896000
- The periodicity of the time series data is Monthly

## UE Benefits Search vs UE Rate 2004-19

### Data Exploration

In [55]:
ue_200419 = pd.read_csv('data/UE Benefits Search vs UE Rate 2004-19.csv')

In [61]:
ue_200419.head()

MONTH  UE_BENEFITS_WEB_SEARCH  UNRATE
0  2004-01                      34     5.7
1  2004-02                      33     5.6
2  2004-03                      25     5.8
3  2004-04                      29     5.6
4  2004-05                      23     5.6

In [67]:
ue_200419.tail()

MONTH  UE_BENEFITS_WEB_SEARCH  UNRATE
176  2018-09                      14     3.7
177  2018-10                      15     3.8
178  2018-11                      16     3.7
179  2018-12                      17     3.9
180  2019-01                      21     4.0

In [57]:
ue_200419.dtypes

MONTH                      object
UE_BENEFITS_WEB_SEARCH      int64
UNRATE                    float64
dtype: object

In [65]:
ue_200419.describe()

UE_BENEFITS_WEB_SEARCH      UNRATE
count              181.000000  181.000000
mean                35.110497    6.217680
std                 20.484925    1.891859
min                 14.000000    3.700000
25%                 21.000000    4.700000
50%                 26.000000    5.400000
75%                 45.000000    7.800000
max                100.000000   10.000000

### Data Exploration Summary

- There are 3 columns with 181 rows of data
- The column data types are: 'MONTH'= object , 'UE_BENEFITS_WEB_SEARCH' = int64. 'UNRATE'=float64
- The data in the columns represent:
'MONTH'= The date , 'UE_BENEFITS_WEB_SEARCH' =  Monthly search interest on Google, 'UNRATE'= The US Unemployment Rate %
- The highest number of searches during the period was 100
- The lowest number of searches during the period was 14
- The highest the US Unemployment Rate was during the period was 10%
- The lowest the US Unemployment Rate was during the period was 3.7%
- The periodicity of the time series data is Monthly

## Bitcoin

### Data Exploration

In [87]:
btc_searches = pd.read_csv('data/Bitcoin Search Trend.csv')
btc_price = pd.read_csv('data/Daily Bitcoin Price.csv')

In [89]:
btc_searches

MONTH  BTC_NEWS_SEARCH
0   2014-09                5
1   2014-10                4
2   2014-11                4
3   2014-12                4
4   2015-01                5
..      ...              ...
68  2020-05               22
69  2020-06               13
70  2020-07               14
71  2020-08               16
72  2020-09               13

[73 rows x 2 columns]

In [93]:
btc_searches.dtypes

MONTH              object
BTC_NEWS_SEARCH     int64
dtype: object

In [95]:
btc_searches.describe()

BTC_NEWS_SEARCH
count        73.000000
mean         15.013699
std          15.146959
min           3.000000
25%           5.000000
50%          14.000000
75%          18.000000
max         100.000000

In [97]:
btc_price

DATE         CLOSE        VOLUME
0     2014-09-17    457.334015  2.105680e+07
1     2014-09-18    424.440002  3.448320e+07
2     2014-09-19    394.795990  3.791970e+07
3     2014-09-20    408.903992  3.686360e+07
4     2014-09-21    398.821014  2.658010e+07
...          ...           ...           ...
2199  2020-09-24  10745.548828  2.301754e+10
2200  2020-09-25  10702.290039  2.123255e+10
2201  2020-09-26  10754.437500  1.810501e+10
2202  2020-09-27  10774.426758  1.801688e+10
2203  2020-09-28  10912.536133  2.122653e+10

[2204 rows x 3 columns]

In [99]:
btc_price.dtypes

DATE       object
CLOSE     float64
VOLUME    float64
dtype: object

In [101]:
btc_price.describe()

CLOSE        VOLUME
count   2203.000000  2.203000e+03
mean    4429.421245  8.043622e+09
std     4148.150071  1.176529e+10
min      178.102997  5.914570e+06
25%      433.629502  6.029915e+07
50%     3637.520020  2.018890e+09
75%     7997.372803  1.322478e+10
max    19497.400391  7.415677e+10

### Data Exploration Summary

- In btc_searches there are 2 columns with 73 rows of data
- The column data types are : 'MONTH' = object, 'BTC_NEWS_SEARCH'= int64
- The data in the columns represent: 'MONTH' - The date, 'BTC_NEWS_SEARCH'= Monthly search interest on Google
- The highest number of searches was 100
- The lowest number of searches was 3
- The periodicity of the time series data is Monthly

- In btc_price there are 3 columns with 2204 rows of data
- The column data types are : 'DATE' = object, 'CLOSE' = float64, 'VOLUME' = float64
- The data in the columns represent : 'DATE' = The date, 'CLOSE' = BTC closing price, 'VOLUME' = BTC Volume Traded
- The highest closing price was 19497.400391
- The lowest closing price was 178.102997
- The highest volume traded was 7.415677e+10
- The lowest volume traded was 5.914570e+06
- The periodicity of the time series data is Daily

## What does the number of Google searches actually mean?

Monthly search interest on Google - Google defines the values of search interest as: 
- 'Numbers represent search interest relative to the highest point on the chart for the given region and time. A value of 100 is the peak popularity for the term. A value of 50 means that the term is half as popular. A score of 0 means there was not enough data for this term. '

The actual search volume of a particular term is not publicly available, what Google offers is a scaled number:
- Each data point is divided by the total searches of the geography and the time range it represents, this is used to compare relative popularity.
- For each word, Google checks how much search volume in each region and time period the word had, relative to ALL the searches in that region and time period. All of these measures are then combined into a single measure of popularity, for which the values are then scaled across the terms provided - the largest value being 100.


## Data Cleaning
- Remove rows containing missing or NaN values
- Fix data types

In [124]:
tesla.isna().value_counts()

MONTH  TSLA_WEB_SEARCH  TSLA_USD_CLOSE
False  False            False             124
Name: count, dtype: int64

In [134]:
ue_200419.isna().value_counts()

MONTH  UE_BENEFITS_WEB_SEARCH  UNRATE
False  False                   False     181
Name: count, dtype: int64

In [144]:
btc_searches.isna().value_counts()

MONTH  BTC_NEWS_SEARCH
False  False              73
Name: count, dtype: int64

In [148]:
btc_price.isna().value_counts()

DATE   CLOSE  VOLUME
False  False  False     2203
       True   True         1
Name: count, dtype: int64

In [158]:
btc_price[btc_price.isna().any(axis=1)]

DATE  CLOSE  VOLUME
2148  2020-08-04    NaN     NaN

Removal of row containing NaN or missing values:

In [165]:
btc_price.dropna(inplace=True)

In [167]:
btc_price[btc_price.isna().any(axis=1)]

Empty DataFrame
Columns: [DATE, CLOSE, VOLUME]
Index: []

In [193]:
tesla.dtypes

MONTH               object
TSLA_WEB_SEARCH      int64
TSLA_USD_CLOSE     float64
dtype: object

In [197]:
tesla['MONTH'] = pd.to_datetime(tesla['MONTH'])

In [199]:
tesla.dtypes

MONTH              datetime64[ns]
TSLA_WEB_SEARCH             int64
TSLA_USD_CLOSE            float64
dtype: object

In [203]:
ue_200419.dtypes

MONTH                      object
UE_BENEFITS_WEB_SEARCH      int64
UNRATE                    float64
dtype: object

In [207]:
ue_200419['MONTH'] = pd.to_datetime(ue_200419['MONTH'])

In [221]:
ue_200419.dtypes

MONTH                     datetime64[ns]
UE_BENEFITS_WEB_SEARCH             int64
UNRATE                           float64
dtype: object

In [223]:
btc_searches.dtypes

MONTH              object
BTC_NEWS_SEARCH     int64
dtype: object

In [225]:
btc_searches['MONTH'] = pd.to_datetime(btc_searches['MONTH'])

In [227]:
btc_searches.dtypes

MONTH              datetime64[ns]
BTC_NEWS_SEARCH             int64
dtype: object

In [231]:
btc_price.dtypes

DATE       object
CLOSE     float64
VOLUME    float64
dtype: object

In [233]:
btc_price['DATE'] = pd.to_datetime(btc_price['DATE'])

In [235]:
btc_price.dtypes

DATE      datetime64[ns]
CLOSE            float64
VOLUME           float64
dtype: object

Resample btc_prices DATE from day to month:

In [238]:
btc_price_monthly = btc_price.resample('M', on='DATE').last()

C:\Users\amitp\AppData\Local\Temp\ipykernel_17648\1577701695.py:1: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  btc_price_monthly = btc_price.resample('M', on='DATE').last()


In [240]:
btc_price_monthly

CLOSE        VOLUME
DATE                                  
2014-09-30    386.944000  3.470730e+07
2014-10-31    338.321014  1.254540e+07
2014-11-30    378.046997  9.194440e+06
2014-12-31    320.192993  1.394290e+07
2015-01-31    217.464005  2.334820e+07
...                  ...           ...
2020-05-31   9461.058594  2.777329e+10
2020-06-30   9137.993164  1.573580e+10
2020-07-31  11323.466797  2.316047e+10
2020-08-31  11680.820313  2.228593e+10
2020-09-30  10912.536133  2.122653e+10

[73 rows x 2 columns]